collectors/HostSystemPropertiesCollector.py

```diff
@@ -0,0 +1,89 @@
+            for property_pair in self.property_yaml['HostSystemPropertiesCollector']['number_metrics']:
+                property_label = property_pair['label']
+                propkey = property_pair['property']
+                values = Resources.get_latest_number_properties_multiple(target, token, uuids, propkey)
+                if not values:
+                    continue
+                for value_entry in values:
+                    data = value_entry['data']
+                    host_id = value_entry['resourceId']
+                    g.add_metric(
+                        labels=[self.hosts[host_id]['datacenter'], self.hosts[host_id]['parent_cluster_name'],
+                                self.hosts[host_id]['name'], property_label],
+                        value=data)
+
+            for property_pair in self.property_yaml["HostSystemPropertiesCollector"]['enum_metrics']:
+                property_label = property_pair['label']
+                propkey = property_pair['property']
+                expected_state = property_pair['expected']
+                values = Resources.get_latest_enum_properties_multiple(target, token, uuids, propkey, expected_state)
+                if not values:
+                    continue
+                for value_entry in values:
+                    data = value_entry['data']
+                    host_id = value_entry['resourceId']
+                    latest_state = value_entry['latest_state']
+                    g.add_metric(
+                        labels=[self.hosts[host_id]['datacenter'], self.hosts[host_id]['parent_cluster_name'],
+                                self.hosts[host_id]['name'], property_label + ": " + latest_state],
+                        value=data)
+
+            for property_pair in self.property_yaml["HostSystemPropertiesCollector"]['info_metrics']:
+                property_label = property_pair['label']
+                propkey = property_pair['property']
+                values = Resources.get_latest_info_properties_multiple(target, token, uuids, propkey)
+                if not values:
+                    continue
+                for value_entry in values:
+                    host_id = value_entry['resourceId']
+                    try:
+                        info_value = float(value_entry['data'])
+                        g.add_metric(
+                            labels=[self.hosts[host_id]['datacenter'], self.hosts[host_id]['parent_cluster_name'],
+                                    self.hosts[host_id]['name'], property_label],
+                            value=info_value)
+                    except ValueError:
+                        info = value_entry['data']
+                        info_value = 0
+                        g.add_metric(
+                            labels=[self.hosts[host_id]['datacenter'], self.hosts[host_id]['parent_cluster_name'],
+                                    self.hosts[host_id]['name'], property_label + ": " + info],
```

Comment:
info could be `'None'` if it is hitting the else condition in Resources. What would be the point of having this set to None?

Reply:
prometheus can"t handle the value of `None`. I'm guessing it must be a number.